aztec/src/main/kotlin/org/wordpress/aztec/AztecTagHandler.kt

```diff
@@ -163,8 +166,8 @@ class AztecTagHandler(val context: Context, val plugins: List<IAztecPlugin> = Ar
             start(output, AztecMediaClickableSpan(mediaSpan))
             output.append(Constants.IMG_CHAR)
         } else {
-            end(output, mediaSpan.javaClass)
             end(output, AztecMediaClickableSpan::class.java)
+            end(output, mediaSpan.javaClass)
```

Comment:
wondering, was this place interchange a bug before?

Reply:
Hah, it wasn't a bug because on the spans-space, there is no meaning to the particular order the spans will be added to the Spannable or not.

But, because now the `start()` and `end()` methods use a LIFO Stack to keep track of the nesting of spans, the out-of-order `end()` calls fail to find the expected tag/span at the top of the Stack.

In this particular case, there are 2 `start` calls when opening the tag (https://github.com/wordpress-mobile/AztecEditor-Android/blob/bd1c9fa37ff5f332720186c3f5d1dee5f4e0944d/aztec/src/main/kotlin/org/wordpress/aztec/AztecTagHandler.kt#L165-L166) where the first one pushes a `AztecMediaSpan` to the stack and the second pushes a `AztecMediaClickableSpan`. So, when ending the tag, we need to call the `end`s in reverse order. That's the change introduced here.

Does that make sense?

Reply:
oh perfect - yes that makes sense 👌 thanks for clarifying!